src/thunderbots/software/grsim_communication/motion_controller.cpp

```diff
@@ -39,19 +40,60 @@ MotionController::Velocity MotionController::bangBangVelocityController(
     }
     else
     {
-        robot_velocities.linear_velocity = MotionController::determineLinearVelocity(
-            robot, dest, desired_final_speed, delta_time, max_speed_meters_per_second,
-            max_angular_speed_radians_per_second);
-        robot_velocities.angular_velocity = MotionController::determineAngularVelocity(
-            robot, desired_final_orientation, delta_time,
-            max_angular_speed_radians_per_second,
-            max_angular_acceleration_meters_per_second_squared);
+        robot_velocities = std::visit(
+            [robot, motion_command, delta_time, max_speed_meters_per_second,
+             max_angular_speed_radians_per_second,
+             max_acceleration_meters_per_second_squared,
+             max_angular_acceleration_meters_per_second_squared](auto&& arg) {
+                using T = std::decay_t<decltype(arg)>;
+
+                MotionController::Velocity robot_velocities;
+
+                if constexpr (std::is_same_v<T, MotionController::PositionCommand>)
+                {
+                    MotionController::PositionCommand command =
+                        std::get<MotionController::PositionCommand>(motion_command);
+                    robot_velocities.linear_velocity =
+                        MotionController::determineLinearVelocityFromPosition(
+                            robot, command.global_destination,
+                            command.final_speed_at_destination, delta_time,
+                            max_speed_meters_per_second,
+                            max_angular_speed_radians_per_second);
+                    robot_velocities.angular_velocity =
+                        MotionController::determineAngularVelocityFromPosition(
+                            robot, command.final_orientation, delta_time,
+                            max_angular_speed_radians_per_second,
+                            max_angular_acceleration_meters_per_second_squared);
+                }
+                else  // Velocity Command
```

Comment:
Just to be future-proof here, can we put the Velocity Command case in an `else if` case, and make the `else` case throw an exception indicating we got an unsupported type?